### PR TITLE
Move theming support from Beta to Main

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #39 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Move theming support from Beta to Main by implementing color theming capabilities for the auro-nav component, fixing a text wrap UI error, and updating documentation to reflect these changes. Introduce custom versioned components to manage dependencies effectively.

New Features:
- Introduce color theming support for the auro-nav component, allowing customization of various UI elements using design tokens.

Bug Fixes:
- Fix text wrap UI error in the auro-nav component to ensure proper display of text.

Enhancements:
- Add custom versioned auro components to manage dependencies and avoid conflicts.

Documentation:
- Update documentation to include new theming support and remove outdated version control recommendations.

### CODE REVIEW

File: docs/partials/api.md [hunk 1]
```diff
@@ -37,3 +37,10 @@ This example shows a user how set up an `auro-nav` component using `auro-anchorl
 <!-- AURO-GENERATED-CONTENT:END -->
 
 </auro-accordion>
+
+### Theme Support
+
+The component may be restyled using the following code sample and changing the values of the following token(s).
+
+<!-- AURO-GENERATED-CONTENT:START (CODE:src=./../../src/tokens.scss) -->
+<!-- AURO-GENERATED-CONTENT:END -->
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply docs/partials/api.md:46]: This is not missing.

File: test/auro-nav.test.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ describe('auro-nav', () => {
 
     await expect(nav.hasAttribute('aria-expanded')).to.be.false;
 
-    const button = nav.shadowRoot.querySelector('auro-button');
+    const button = nav.shadowRoot.querySelector('.showHideBtn');
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-nav.test.js:57]: Not implementing at this time.

File: src/style-anchorlink.scss [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
 @import './../node_modules/@aurodesignsystem/webcorestylesheets/src/core';
 
-/* stylelint-disable declaration-empty-line-before, declaration-no-important, order/properties-order, max-line-length */
-
 :host {
   display: block;
-  border: 1px solid transparent;
-  border-left: 1px solid var(--ds-color-border-divider-default, $ds-color-border-divider-default);
+  
+  border-width: 1px;
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply src/style-anchorlink.scss:17]: Not implementing at this time.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**